src/components/user/Countdown.jsx

```diff
@@ -1,5 +1,85 @@
+import React, { useState, useEffect } from "react";
+import data from "/src/data/Config.js";
+
+const parts = data.end_date.split(" ");
+const day = parseInt(parts[1]);
+const month = parts[0];
+const year = parseInt(parts[2]);
+const monthIndex = new Date(Date.parse(month + " 1, " + year)).getMonth() + 1;
+const endDate = new Date(year, monthIndex - 1, day, 23, 59, 59);
+
 const Countdown = () => {
-  return <div>Countdown</div>;
+  const [countdown, setCountdown] = useState({
+    days: 0,
+    hours: 0,
+    minutes: 0,
+    seconds: 0,
+  });
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      const now = new Date().getTime();
+      const timeLeft = endDate - now;
+      const days = Math.floor(timeLeft / (1000 * 60 * 60 * 24));
+      const hours = Math.floor(
+        (timeLeft % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60)
+      );
+      const minutes = Math.floor((timeLeft % (1000 * 60 * 60)) / (1000 * 60));
+      const seconds = Math.floor((timeLeft % (1000 * 60)) / 1000);
+      timeLeft <= 0
+        ? setCountdown({ days: 0, hours: 0, minutes: 0, seconds: 0 })
+        : setCountdown({ days, hours, minutes, seconds });
+    }, 1000);
+    return () => clearInterval(interval);
+  }, []);
+
+  return (
+    <div className="inline-block bg-green-400 mr-3 mb-3 rounded-lg font-bold">
```

Comment:
check the implementation of the countdowns in bearhack and designverse, they map through the digit UI elements to minimize the code